libraries/AP_MSP/AP_MSP_Telem_DisplayPort.cpp

```diff
@@ -41,4 +41,16 @@ MSPCommandResult AP_MSP_Telem_DisplayPort::msp_process_out_fc_variant(sbuf_t *ds
     return MSP_RESULT_ACK;
 }
 
+bool AP_MSP_Telem_DisplayPort::init_uart()
+{
+    if (_msp_port.uart != nullptr)  {
+        // re-init port here for use in this thread
+        // displayport needs a bigger tx buffer
+        // rx buffer can be small for we mostly write
+        _msp_port.uart->begin(0, 256, 768);
```

Comment:
As a note this doesn't actually shrink the RX size on any of the ChibiOS flight controllers. The ChibiOS HAL has a default minimum buffer size of 512 and we won't go below that except in a couple of situations. You can work around this in the hwdef, but at the moment I believe that's only done for some of the AP_Periph boards.

Reply:
You're absolutely right, RX min is 512, this patch does reduce available memory

Reply:
It's not a blocker from my point of view, but I was highlighting it since the same memory footprint was originally targeted. I'm not sure how constrained these boards are on memory.

Reply:
I could add an enabler bit somewhere, like in MSP_OPTIONS to extend the buffer on demand?

Reply:
> As a note this doesn't actually shrink the RX size on any of the ChibiOS flight controllers. The ChibiOS HAL has a default minimum buffer size of 512 and we won't go below that except in a couple of situations. You can work around this in the hwdef, but at the moment I believe that's only done for some of the AP_Periph boards.

I never knew that and I think it's daft that this is true! We're probably wasting tons of space on F4. I wonder if this affects the timing of other things ...